stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,534 @@
+        :param pickings: stock.picking recordset
+        :return: dict
+        """
+        picking = fields.first(pickings)
+        partner_id = picking._get_partner_to_invoice()
+        partner = self.env["res.partner"].browse(partner_id)
+        inv_type = self._get_invoice_type()
+        currency = self.env.company.currency_id
+        if partner:
+            code = picking.picking_type_id.code
+            if partner.property_product_pricelist and code == "outgoing":
+                currency = partner.property_product_pricelist.currency_id
+        journal = self._get_journal()
+        invoice_obj = self.env["account.move"]
+        values = invoice_obj.default_get(invoice_obj.fields_get().keys())
+        values.update(
+            {
+                "invoice_origin": ", ".join(pickings.mapped("name")),
+                "partner_id": partner_id,
+                "move_type": inv_type,
+                "currency_id": currency.id,
+                "journal_id": journal.id,
+            }
+        )
+
+        invoice, values = self._simulate_invoice_onchange(values)
+        return invoice, values
+
+    def _get_move_key(self, move):
+        """
+        Get the key based on the given move
+        :param move: stock.move recordset
+        :return: (key,)
+        """
+        key = (move,)
+        if self.group == "partner_product":
+            key = (move.product_id,)
+        return key
+
+    def _group_moves(self, moves):
+        """
+        Possibility to group moves (to create 1 invoice line with many moves)
+        :param moves: stock.move recordset
+        :return: list of stock.move recordset
+        """
+        grouped_moves = {}
+        moves = moves.filtered(lambda m: m.invoice_state == "2binvoiced")
+        for move in moves:
+            key = self._get_move_key(move)
+            move_grouped = grouped_moves.get(key, self.env["stock.move"].browse())
+            move_grouped |= move
+            grouped_moves.update({key: move_grouped})
+        return grouped_moves.values()
+
+    def _simulate_invoice_line_onchange(self, values, price_unit=None):
+        """
+        Simulate onchange for invoice line
+        :param values: dict
+        :return: dict
+        """
+        line = self.env["account.move.line"].new(values.copy())
+        line._inverse_partner_id()
+        line._inverse_product_id()
+        line._inverse_account_id()
+        line._inverse_amount_currency()
+        new_values = line._convert_to_write(line._cache)
+        if price_unit:
+            new_values["price_unit"] = price_unit
+        # Ensure basic values are not updated
+        values.update(new_values)
+        return values
+
+    def _get_invoice_line_values(self, moves, invoice_values, invoice):
+        """
+        Create invoice line values from given moves
+        :param moves: stock.move
+        :param invoice: account.move
+        :return: dict
+        """
+        name = ", ".join(moves.mapped("name"))
```

Comment:
hi, @mbcosta we found a problem when invoicing a picking with multiple moves with the same product it duplicates the name of the product in the invoice line and if we have the group = "partner_product" . We think that if changing this line to only get the first move's description if using the partner_prooduct option should avoid the issue.